### ISSUE TITLE
Support for DSMR-firmware v5.0

### ISSUE THREAD

**Issue by dennissiemensma** [dennissiemensma/dsmr-reader#221]
It seems that DSMR v5 is rolled out now (#212). It should have a unittest as well.

As I don't have one at home, I'm relying on one being provided.

* Check CRC.
* Check values.
* Check connection parameters.
* Allow gas readings to be taken each minute (grouped) as well.
* Serial connection should either **stay open** or delay for 5 seconds.

**Comment by dennissiemensma** [comment]
From the DSMR v5.0.2 docs:

```
The interface must use a fixed transfer speed of 115200 baud.

The Metering System must send its data to the OSM device every single second and the
transmission of the entire P1 telegram must be completed within 1s.

The format of transmitted data must be defined as “8N1”. Namely:
- 1 start bit,
- 8 data bits,
- no parity bit and
- 1 stop bit.
```

Code settings for v4+

```
    serial_handle.baudrate = 115200
    serial_handle.xonxoff = 1
    serial_handle.bytesize = serial.EIGHTBITS
    serial_handle.parity = serial.PARITY_NONE,
    serial_handle.stopbits = serial.STOPBITS_ONE
```

So that seems to be ok.

**Comment by dennissiemensma** [comment]
I was finally able to commit some progress. It's quite hard to keep supporting the 'legacy' DSMR v4, both grouped and ungrouped, while also allowing unrestricted data for DSMR v5. And that version has be to compatible with grouping and non-grouping as well.

It still needs to be tested quite well, but the unittests seem to pass in the way I asume it should work.

**Comment by dennissiemensma** [comment]
I'll run this one my second evironment for some time to see whether the way gas readings are stored is unaffected by this (big) change.

**Comment by dennissiemensma** [comment]
No differences between my two production environments (v1.5.5 / v1.6b), so seems good. Too bad I cannot test DSMR v5 here.

**Comment by dennissiemensma** [comment]
It seems that the statistics aggregate their data by range, which works for DSMR v5 as well. So I think this can be marked completed now.

Merged back to v1.6 branch.
